src/test/java/com/sap/cloudfoundry/client/facade/CloudControllerClientIntegrationTest.java

```diff
@@ -0,0 +1,125 @@
+        client = new CloudControllerClientImpl(apiUrl, credentials, target, true);
+
+    }
+
+    @BeforeEach
+    public void setUp(TestInfo testInfo) {
+        System.out.println("================================");
+        System.out.println(String.format("Test started: %s", testInfo.getDisplayName()));
+        sequential.lock();
+    }
+
+    @AfterEach
+    public void tearDown(TestInfo testInfo) {
+        sequential.unlock();
+        System.out.println(String.format("Test finished: %s", testInfo.getDisplayName()));
+    }
+
+    @Test
+    @DisplayName("Adding missing domain")
+    void addDomain() throws IOException {
+        String domainName = ITVariable.DOMAIN_NAME.getValue();
+        assertDomainExists(domainName, false);
+        try {
+            client.addDomain(domainName);
+            assertDomainExists(domainName, true);
+        } finally {
+            client.deleteDomain(domainName);
+        }
+    }
+
+    @Test
+    @DisplayName("Adding existing domain")
+    void addDomainAlreadyExists() throws IOException {
+        String domainName = ITVariable.DOMAIN_NAME.getValue();
+        client.addDomain(domainName);
+        try {
+            assertDomainExists(domainName, true);
+            client.addDomain(domainName);
+        } finally {
+            client.deleteDomain(domainName);
+        }
+    }
+
+    @Test
+    @DisplayName("Delete existing domain")
+    void deleteDomain() throws IOException {
+        String domainName = ITVariable.DOMAIN_NAME.getValue();
+        client.addDomain(domainName);
+        assertDomainExists(domainName, true);
+        client.deleteDomain(domainName);
+        assertDomainExists(domainName, false);
+    }
+
+    @Test
+    @DisplayName("Delete missing domain")
+    void deleteDomainMissing() throws IOException {
+        String domainName = ITVariable.DOMAIN_NAME.getValue();
+        assertDomainExists(domainName, false);
+        try {
+            client.deleteDomain(domainName);
+            fail();
+        } catch (CloudOperationException e) {
+            assertEquals(HttpStatus.NOT_FOUND, e.getStatusCode());
+        }
+
+    }
+
+    private static void assertAllRequiredVariablesAreDefined() {
+        for (ITVariable itVariable : ITVariable.values()) {
+            Assertions.assertNotNull(itVariable.getValue(),
+                                     String.format("Missing required value defined by env var %s or system property %s",
+                                                   itVariable.getEnvVariable(), itVariable.getProperty()));
+        }
+    }
+
+    private static void assertDomainExists(String domainName, boolean domainExists) {
+        boolean actualDomainExists = client.getDomainsForOrganization()
+                                           .stream()
+                                           .map(CloudDomain::getName)
+                                           .anyMatch(name -> StringUtils.equals(domainName, name));
```

Comment:
I am not really sure about that proposal and it is not needed at all but:
`.anyMatch(domainName::equals));`